sstables/sstables.cc

```diff
@@ -2143,6 +2143,16 @@ future<> sstable::move_to_quarantine(bool do_sync_dirs) {
     co_await move_to_new_dir(std::move(new_dir), generation(), do_sync_dirs);
 }
 
+future<shared_sstable> sstable::clone_at(const sstring& new_dir) {
+    if (fs::canonical(fs::path(new_dir)) == fs::canonical(fs::path(_dir))) {
+        on_internal_error(sstlog, format("Cannot clone sstable {} into same dir", get_filename()));
```

Comment:
Does this really warrant an abort?

Reply:
We can't and don't have any use case for cloning a sstable into the same dir with the same generation.
It is a software bug that should be found and fixed in dev/test cycles - i.e. akin to assert.

Reply:
Kept in v5